packages/grafana-ui/src/components/ShareTimeRangeButton/ShareTimeRangeButton.tsx

```diff
@@ -0,0 +1,110 @@
+import { css } from '@emotion/css';
+import { useRef, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+import { copyText } from '../../utils/clipboard';
+import { t, Trans } from '../../utils/i18n';
+import { absoluteTimeRangeURL } from '../../utils/time';
+import { Button, ButtonGroup, ButtonProps } from '../Button';
+import { ClipboardButton } from '../ClipboardButton/ClipboardButton';
+import { Dropdown } from '../Dropdown/Dropdown';
+import { Menu } from '../Menu/Menu';
+import { MenuItemElement } from '../Menu/MenuItem';
+
+export interface Props extends ButtonProps {
+  /**
+   * Whether to collapse the button text
+   */
+  collapsed?: boolean;
+
+  /**
+   * The URL to share
+   */
+  url?: string;
+
+  /**
+   * The from parameter to use in the URL
+   *
+   * @default 'from'
+   */
+  fromParam?: string;
+
+  /**
+   * The to parameter to use in the URL
+   *
+   * @default 'to'
+   */
+  toParam?: string;
+}
+
+export function ShareTimeRangeButton({ collapsed, url: urlProp, fromParam, toParam }: Props) {
+  const [isOpen, setIsOpen] = useState(false);
+  const styles = useStyles2(getStyles);
+  const url = urlProp ?? window.location.href;
+
+  const relativeUrlRef = useRef<MenuItemElement>(null);
+  const absoluteUrlRef = useRef<MenuItemElement>(null);
+
+  const clickHandler = (text: string, ref: React.RefObject<MenuItemElement>) => {
+    copyText(text, ref);
+    setIsOpen(false);
+  };
+
+  const menu = (
+    <Menu>
+      <Menu.Item
+        key="copy-url-relative"
```

Comment:
nit: `key` is unnecessary for these items. it's only needed if you're mapping over an array, for example